app/news-page/stores/edit-post.store.ts

```diff
@@ -0,0 +1,70 @@
+import { FormState } from 'formstate';
+import { action, observable } from 'mobx';
+
+import { formStateToJS, InputFieldState, TextAreaFieldState } from '../../common/utils/form-helpers';
+import { FormValidators } from '../../common/utils/form-validators';
+import { NewsApi } from '../api/news.api';
+import { Post } from '../models/post';
+
+export class EditPostStore {
+    result: Promise<Post | undefined>;
+    private close!: (saved?: Post) => void;
```

Comment:
The news component should decide when to close the form.

Reply:
`NewsStore` decides this with `this.editPostStore = void 0;`